e2e/test-project-initialization.sh

```diff
@@ -45,17 +58,28 @@ for pkg_manager in $pkg_managers; do
 done
 
 for pkg_manager in $pkg_managers; do
-  echo "\n\n[e2e] Running tests with package manager: $pkg_manager"
+  pkg_runner=$pkg_manager
+  if [ "$pkg_manager" = "npm" ]; then
+    pkg_runner="npx"
+  fi
+
+  if [ "$pkg_manager" = "pnpm" ] && [ "$IS_WINDOWS" = "true" ]; then
+    # TODO: There is a bug with pnpm on Windows; the HH installation runs twice. Skip pnpm on Windows for the moment.
+    continue
+  fi
+
+  printf "\n\n[e2e] Running tests with package manager '$pkg_manager' and package runner '$pkg_runner'\n"
 
   # pkg_manager, javascript, cjs
   echo "[e2e] Testing: $pkg_manager, javascript, cjs"
   mkdir ${TESTS_DIR}/${pkg_manager}-javascript-cjs
   cd ${TESTS_DIR}/${pkg_manager}-javascript-cjs
   create_package_json
-  $pkg_manager add ../../../packages/hardhat-core/$HARDHAT_TGZ_FILE >/dev/null 2>&1
-  HARDHAT_CREATE_JAVASCRIPT_PROJECT_WITH_DEFAULTS=true npx hardhat init
-  npx hardhat compile
-  npx hardhat test
+  $pkg_manager add $HARDHAT_CORE_FOLDER_PATH/$HARDHAT_TGZ_FILE >/dev/null 2>&1
```

Comment:
These lines are duped several times. Maybe it's worth it to move them to a fn?

Reply:
I'm ok with this duplication (for now), because bash is so bad for abstracting things, and I'd rather have more straightforward code. But I'm fine either way.